### DISCUSSION TITLE
Output the energy contribution for different terms

### DISCUSSION THREAD

**Discussion by emmelines** [idaholab/moose#16977] [Q&A Modules: Phase field] [answered]
Hi,

I set up a simulation base on the Phase-field model which combines the Cahn-Hillard and Allen-Cahn equation.
And now I would like to see the free energy change for only one term(Like composition only) How should I set it up?
Thank you.

best,

Emmeline

**Comment by jessecarterMOOSE** [comment, marked as answer]
Perhaps you could use the [`TotalFreeEnergy`](https://mooseframework.inl.gov/source/auxkernels/TotalFreeEnergy.html) AuxKernel and just pass it your concentration variable and its gradient coefficient. Then integrate the AuxVariable with a postprocessor to see the total energy change.
